src/data/auroradb.ts

```diff
@@ -0,0 +1,249 @@
+}
+
+interface NaptanInfo {
+    commonName: string;
+    naptanCode: string;
+    atcoCode: string;
+    nptgLocalityCode: string;
+    localityName: string;
+    parentLocalityName: string;
+    indicator: string;
+    street: string;
+}
+
+interface NaptanAtcoCodes {
+    naptanCode: string;
+    atcoCode: string;
+}
+
+export interface Stop {
+    stopName: string;
+    naptanCode: string;
+    atcoCode: string;
+    localityCode: string;
+    localityName: string;
+    parentLocalityName: string;
+    indicator?: string;
+    street?: string;
+    qualifierName?: string;
+}
+
+export interface StopIdentifiers {
+    naptanCode: string | null;
+    atcoCode: string;
+}
+
+export interface Service {
+    serviceDescription: string;
+    operatorShortName: string;
+    journeyPatterns: JourneyPattern[];
+}
+
+export interface RawService {
+    serviceDescription: string;
+    operatorShortName: string;
+    journeyPatterns: RawJourneyPattern[];
+}
+
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+const getAuroraDBClient = () => {
+    let client = null;
+
+    if (process.env.NODE_ENV === 'development') {
+        client = createConnection({
+            host: 'mysql',
+            user: 'fdbt_site',
+            password: 'password',
+            database: 'fdbt',
+        });
+    } else {
+        // TODO find replacement here
+        client = createConnection({
+            host: 'mysql',
+            user: 'fdbt_site',
+            password: 'password',
+            database: 'fdbt',
+        });
+    }
+
+    return client;
+};
+
+export const convertDateFormat = (startDate: string): string => {
+    return dateFormat(startDate, 'dd/mm/yyyy');
+};
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const executeQuery = async (query: string, values: any[]) => {
+    const connection = await getAuroraDBClient();
+    const [rows] = await connection.execute(query, values);
+    connection.end();
+    return rows;
+};
+
+export const getServicesByNocCode = async (nocCode: string): Promise<ServiceType[]> => {
+    try {
+        const queryInput = 'SELECT * FROM `tndsService` WHERE `nocCode` = ?';
+
+        const queryResult = await executeQuery(queryInput, [nocCode]);
+
+        const resultRow = JSON.parse(JSON.stringify(queryResult));
+
+        return (
+            resultRow?.map((item: ServiceType) => ({
+                lineName: item.lineName,
+                startDate: convertDateFormat(item.startDate),
+                description: item.description,
+            })) || []
+        );
+    } catch (err) {
+        throw new Error(`Could not retrieve services from DynamoDB: ${err.name}, ${err.message}`);
+    }
+};
+
+export const batchGetStopsByAtcoCode = async (atcoCodes: string[]): Promise<Stop[] | []> => {
+    let orQuery = '';
+
+    try {
+        atcoCodes.forEach((code, index) => {
+            orQuery += `atcoCode = ${code} ${index < atcoCodes.length - 1 ? 'OR ' : ''}`;
```

Comment:
Was originally using 'IN' statement but was a bit fiddly so ended up constructing or statement. Issue here which i referred to https://github.com/sidorares/node-mysql2/issues/196

Reply:
What was wrong with the `IN` query? Could we have something like:
```
const substitution = atcoCodes.map(() => '?').join(',');
const batchQuery = `SELECT * FROM naptanStop WHERE atcoCode IN (${substitution})`;

const results = await executeQuery(batchQuery, atcoCodes);
```